packages/dialect-react-ui/components/Chat/screens/Main/pages/CreateThreadPage/CreateThread.tsx

```diff
@@ -105,6 +129,36 @@ export default function CreateThread({
       });
   };
 
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(async () => {
+      if (address.length === 0) {
```

Comment:
I would always recommend to avoid nesting:

```
if (!address) {
  setValidAddress(false);
  return;
}

if (address.charAt(0) !== '@') {
  try {...} catch {...}
  return;
}

const handle = ...;
...
```

Hope this makes sense. Less nesting = better readability

Reply:
agreed and updated to remove the nesting